metafetcher/fetcher.go

```diff
@@ -0,0 +1,136 @@
+package metafetcher
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/NebulousLabs/skynet-accounts/database"
+
+	"github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// maxAttempts defines the maximum number of attempts we are going to make to
+// process a message.
+const maxAttempts = 3
+
+// Message is the format we use to tell the MetaFetcher to download
+// the metadata for a given skylink and then add its size to the used space of
+// a given user.
+type Message struct {
+	UserID    primitive.ObjectID
+	SkylinkID primitive.ObjectID
+	Attempts  uint8
+}
+
+// MetaFetcher is a background task that listens for messages on its queue and
+// then processes them.
+type MetaFetcher struct {
```

Comment:
(nit) why abbreviate to `Meta` - I would call it SkylinkMetadataFetcher to make it super obvious

Reply:
Because I hate long lines, haha.
I don't think that the extra characters there would resolve any ambiguity or bring anything useful to the table.